src/components/App.jsx

```diff
@@ -0,0 +1,135 @@
+import { useEffect, useState } from 'react';
+
+import Header from './Header.jsx';
+import Main from './Main.jsx';
+import Sidebar from './Sidebar.jsx';
+import LoginModal from './LoginModal.jsx';
+import SignupModal from './signupModal.jsx';
+import PreviewModal from './PreviewModal.jsx';
+import GoogleBooksSearch from './Views/ActiveBooklistView.jsx';
+import { getBook, getBestSellers } from '../utils/api.jsx';
+
+import {
+  screenWidthContext,
+  searchResultsContext,
+} from '../contexts/contexts.js';
+
+export default function App() {
+  const [screenWidth, setScreenWidth] = useState(window.innerWidth);
+  const [searchResults, setSearchResults] = useState([]);
+  const [activeModal, setActiveModal] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedBook, setSelectedBook] = useState({});
+  const [shelves, setShelves] = useState([]);
+
+  const handleCloseModal = () => {
+    setActiveModal('');
+  };
+
+  const handleLogin = () => {
+    console.log('logged in');
+  };
+
+  const handleSignup = () => {
+    console.log('signed up');
+  };
+
+  const handleSignupClick = () => {
+    setActiveModal('signup');
+  };
+
+  const handleLoginClick = () => {
+    setActiveModal('login');
+  };
+
+  const handleCardClick = (book) => {
+    console.log(book);
+    setActiveModal('preview');
+    setSelectedBook(book);
+  };
+
+  const toPascalCase = (str) => {
+    return str
+      .toLowerCase()
+      .split(/[\s-_]+/)
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+      .join(' ');
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', (e) => {
+      setScreenWidth(e.target.innerWidth);
+    });
+  }, []);
+
+  //getBook();
+  //getBestSellers();
+
+  useEffect(() => {
+    getBestSellers().then((data) => {
+      const shelvesData = data.results.lists.map((shelf) => {
+        let modifiedShelf = {
+          shelf: `${shelf.display_name}`,
+          rounded: false,
+          cards: shelf.books.map((book) => {
+            //console.log(book)
+            const bookData = {
+              title: toPascalCase(book.title),
+              subtitle: book.author,
+              img: book.book_image,
+              des: book.description,
+              link: book.amazon_product_url,
+            };
+            return bookData;
+          }),
+        };
+        return modifiedShelf;
+      });
+      setShelves(shelvesData);
+    });

```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**